llvm/lib/Target/AIE/AIEHazardRecognizer.cpp

```diff
@@ -324,7 +324,8 @@ AIEHazardRecognizer::getHazardType(SUnit *SU, int DeltaCycles) {
       // Check if there is NoHazard, If there is a Hazard or NoopHazard check
       // for the next possible Opcode.
       if (Haz == NoHazard) {
-        SelectedAltOpcodes[MI] = AltInstOpcode;
+        SelectedAltDescs.setAlternateDescriptor(
+            MI, const_cast<llvm::MCInstrDesc *>(&TII->get(AltInstOpcode)));
```

Comment:
I think we can have a clean call here, like:

```
SelectedAltDescs.setAlternateDescriptor(MI, AltInstOpcode);
```

In this way, TII can be a field in `AIEAlternateDescriptor`.

Reply:
Have updates as suggested. Thank you!